includes/integrations/class-sst-deposits-for-wc.php

```diff
@@ -0,0 +1,658 @@
+				'woocommerce_calculated_total',
+				array( $this, 'set_post_data' )
+			);
+			add_filter(
+				'woocommerce_calculated_total',
+				array( $this, 'fix_cart_deposit_info' ),
+				1110,
+				2
+			);
+			add_filter(
+				'dfw_total_cart_amount_price_figure_html',
+				array( $this, 'fix_total_cart_amount' )
+			);
+		}
+
+		// ORDER HOOKS
+		add_filter(
+			'sst_should_calculate_order_tax',
+			array( $this, 'should_calculate_order_tax' ),
+			10,
+			3
+		);
+		add_filter(
+			'wootax_order_packages',
+			array( $this, 'filter_order_packages' ),
+			10,
+			2
+		);
+		// Priority set to run before DFW_Manage_Orders::dfw_order_init_action()
+		add_action( 'admin_init', array( $this, 'add_gmt_offset_filter' ), 5 );
+		add_action(
+			'woocommerce_new_order',
+			array( $this, 'add_woocommerce_calc_taxes_filter' ),
+			10,
+			2
+		);
+		add_action(
+			'woocommerce_order_before_calculate_totals',
+			array( $this, 'remove_pre_option_filters' ),
+			10,
+			2
+		);
+		add_action(
+			'woocommerce_before_order_object_save',
+			array( $this, 'calculate_order_totals' )
+		);
+		add_action(
+			'woocommerce_after_order_object_save',
+			array( $this, 'adjust_stripe_invoice_amount' )
+		);
+	}
+
+	/**
+	 * Calculate taxes on amount due in future.
+	 */
+	protected function calculate_future_taxes() {
+		$filter_packages = function( $packages ) {
+			return array_map(
+				function( $package ) {
+					// Remove shipping and fees (collected with deposit)
+					if ( $package['shipping'] ) {
+						$package['shipping']->cost = 0;
+					}
+
+					$package['fees'] = array();
+
+					foreach ( $package['contents'] as $key => &$item ) {
+						if ( empty( $item['has_deposit'] ) ) {
+							// Remove items with no deposit
+							unset( $package['contents'][ $key ] );
+							continue;
+						}
+
+						// Update total to reflect amount due in future
+						$item['line_total'] =
+							$item['quantity'] * ( $item['full_amount'] - $item['deposit'] );
+					}
+
+					return $package;
+				},
+				$packages
+			);
+		};
+
+		add_filter( 'wootax_cart_packages', $filter_packages );
+		add_filter( 'wootax_save_packages_for_capture', '__return_false' );
+
+		WC()->cart->calculate_totals();
+
+		remove_filter( 'wootax_cart_packages', $filter_packages );
+		remove_filter( 'wootax_save_packages_for_capture', '__return_false' );
+
+		return array_reduce(
+			WC()->cart->get_cart_contents(),
+			function( $taxes, $item ) {
+				$taxes[ $item['key'] ] = $item['line_tax'];
+				return $taxes;
+			},
+			[]
+		);
+	}
+
+	/**
+	 * Adds tax to the `full_amount` and product price for each item with
+	 * a deposit to match Deposits for WooCommerce's expectations.
+	 *
+	 * @param array $cart_contents WC cart contents
+	 *
+	 * @return float
+	 */
+	public function add_tax_to_totals( $cart_contents ) {
+		$is_paying_in_full = true;
+
+		foreach ( $cart_contents as $item ) {
+			if ( ! empty( $item['has_deposit'] ) ) {
+				$is_paying_in_full = false;
+				break;
+			}
+		}
```

Comment:
There's a DFW helper method to check if any cart item has a deposit. Can probably leverage that here.

Reply:
Can't use it here as it results in infinite recursion. Will stick to this.